pkg/frontend/middleware/validate.go

```diff
@@ -48,50 +48,50 @@ func Validate(env env.Interface, apis map[string]*api.Version) func(http.Handler
 			if _, found := vars["subscriptionId"]; found {
 				_, err := uuid.FromString(vars["subscriptionId"])
 				if err != nil {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", vars["subscriptionId"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidSubscriptionID, "", "The provided subscription identifier '%s' is malformed or invalid.", vars["subscriptionId"])
 					return
 				}
 			}
 
 			if _, found := vars["resourceGroupName"]; found {
 				if !rxResourceGroupName.MatchString(vars["resourceGroupName"]) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeResourceGroupNotFound, "", "Resource group '%s' could not be found.", vars["resourceGroupName"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeResourceGroupNotFound, "", "Resource group '%s' could not be found.", vars["resourceGroupName"])
 					return
 				}
 			}
 
 			if _, found := vars["resourceProviderNamespace"]; found {
 				if vars["resourceProviderNamespace"] != strings.ToLower(resourceProviderNamespace) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidResourceNamespace, "", "The resource namespace '%s' is invalid.", vars["resourceProviderNamespace"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceNamespace, "", "The resource namespace '%s' is invalid.", vars["resourceProviderNamespace"])
 					return
 				}
 			}
 
 			if _, found := vars["resourceType"]; found {
 				if vars["resourceType"] != strings.ToLower(resourceType) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
 					return
 				}
 			}
 
 			if _, found := vars["resourceName"]; found {
 				if !rxResourceGroupName.MatchString(vars["resourceName"]) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s/%s' under resource group '%s' was not found.", vars["resourceProviderNamespace"], vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s/%s' under resource group '%s' was not found.", vars["resourceProviderNamespace"], vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
```

Comment:
if asking for the resource this is a 404.

Reply:
this is validation of the resourceName